clang/test/compiler-course/kurakin_user_data_type/kurakin_user_data_type_test.cpp

```diff
@@ -0,0 +1,83 @@
+// RUN: %clang_cc1 -load %llvmshlibdir/UserDataTypePlugin_Kurakin_Matvey_FIIT1_ClangAST%pluginext -plugin UserDataTypePlugin_Kurakin_Matvey_FIIT1_ClangAST -fsyntax-only %s 2>&1 | FileCheck %s
+
+//CHECK: Human
+//CHECK-NEXT: |_Fields
+//CHECK-NEXT: | |_ age (unsigned int|public)
+//CHECK-NEXT: | |_ height (unsigned int|public)
+//CHECK-NEXT: |
+//CHECK-NEXT: |_Methods
+//CHECK-NEXT: | |_ sleep (void()|public|virtual|pure)
+//CHECK-NEXT: | |_ eat (void()|public|virtual|pure)
+
+//CHECK: Engineer -> Human
+//CHECK-NEXT: |_Fields
+//CHECK-NEXT: | |_ salary (unsigned int|public)
+//CHECK-NEXT: |
+//CHECK-NEXT: |_Methods
+//CHECK-NEXT: | |_ sleep (void()|public|override)
+//CHECK-NEXT: | |_ eat (void()|public|override)
+//CHECK-NEXT: | |_ work (void()|public)
+
+//CHECK: A
+//CHECK-NEXT: |_Fields
+//CHECK-NEXT: | |_ a1 (int|private)
+//CHECK-NEXT: | |_ a2 (float|private)
+//CHECK-NEXT: |
+//CHECK-NEXT: |_Methods
+
+//CHECK: B
+//CHECK-NEXT: |_Fields
+//CHECK-NEXT: |
+//CHECK-NEXT: |_Methods
+//CHECK-NEXT: | |_ b1 (void(int,int)|private|virtual)
+
+//CHECK: C -> A,B
+//CHECK-NEXT: |_Fields
+//CHECK-NEXT: |
+//CHECK-NEXT: |_Methods
+//CHECK-NEXT: | |_ b1 (void(int,int)|private|override)
+
+//CHECK: D
+//CHECK-NEXT: |_Fields
+//CHECK-NEXT: | |_ d1 (int[5]|private)
+//CHECK-NEXT: | |_ d2 (T|protected)
+//CHECK-NEXT: |
+//CHECK-NEXT: |_Methods
+//CHECK-NEXT: | |_ d3 (T(T,int)|public|virtual|pure)
+
+struct Human {
+  unsigned age;
+  unsigned height;
+  virtual void sleep() = 0;
+  virtual void eat() = 0;
+};
+
+struct Engineer : Human {
+  unsigned salary;
+  void sleep() override { /* something */ }
+  void eat() override { /* something */ }
+  void work() { /* something */ }
+};
+
+class A {
+  int a1;
+  float a2;
+};
+
+class B {
+  virtual void b1(int b2, int b3);
+};
+
+class C : A,B{
+  void b1(int b2, int b3) override;
+};
```

Comment:
Please also add the following modifications in your test:
```suggestion
  float a2;
  int somefunc(float);
};

class B {
  virtual void b1(int b2, int b3);
};

class C : A,B{
  void b1(int b2, int b3) override;
  int somefunc(float);
};
```